### ISSUE TITLE
Unable to spread Video props without flow complaining

### ISSUE THREAD

**Issue by keyworks** [pinterest/gestalt#305]
I was trying to implement a wrapper component for Video but could not properly flow type the props of wrapper component due to the following issue:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ app/common/react/components/video/VideoWithContextLogging.js:19:10

Could not decide which case to select. Since case 1 [1] may work but if it doesn't case 2 [1] looks promising too. To
fix add a type annotation to .accessibilityMaximizeLabel [2], to .accessibilityMinimizeLabel [2], to
.accessibilityMuteLabel [2], to .accessibilityPauseLabel [2], to .accessibilityPlayLabel [2], to
.accessibilityUnmuteLabel [2], or to .controls [2].

     app/common/react/components/video/VideoWithContextLogging.js
     16│ }
     17│
     18│ function VideoWrapper(props: React$ElementProps<typeof Video>) {
 [2] 19│   return <Video {...props} />;
     20│ }
     21│
     22│ // class VideoWithContextLogging extends Component<$Shape<Props>> {

     node_modules/gestalt/src/Video.js
 [1] 34│ type Props = {|
     35│   aspectRatio: number,
     36│   captions: string,
     37│   loop?: boolean,
     38│   onDurationChange?: ({
     39│     event: SyntheticEvent<HTMLVideoElement>,
     40│     duration: number,
     41│   }) => void,
     42│   onEnded?: ({ event: SyntheticEvent<HTMLVideoElement> }) => void,
     43│   onFullscreenChange?: ({ event: Event, fullscreen: boolean }) => void,
     44│   onLoadedChange?: ({
     45│     event: SyntheticEvent<HTMLVideoElement>,
     46│     loaded: number,
     47│   }) => void,
     48│   onPlay?: ({ event: SyntheticEvent<HTMLDivElement> }) => void,
       :
     54│     time: number,
     55│   }) => void,
     56│   onVolumeChange?: ({
     57│     event: SyntheticEvent<HTMLDivElement>,
     58│     volume: number,
     59│   }) => void,
     60│   playbackRate: number,
     61│   playing: boolean,
     62│   playsInline?: boolean,
     63│   poster?: string,
     64│   preload: 'auto' | 'metadata' | 'none',
     65│   src: Source,
     66│   volume: number,
     67│   ...Controls,
     68│ |};
```

As workaround to `React$ElementProps<typeof Video>` you can use `*` type but flow does not infer props properly and types it as 'empty'.
```
function VideoWrapper(props: *) {
  return <Video {...props} />;
}
``` 

The issue seems to be that flow cannot determine whether it wants to type it as VideoWithControls or VideoNoControls in Video.js - specifically in this code (editted for brevity):

```
type Controls = VideoWithControls | VideoNoControls;

type Props = {
  ...Controls
}
```

Toy example: https://flow.org/try/#0C4TwDgpgBAaglgEwgewOp2ACwMLIHbABOyANgM5QC8UA3gD4BQUzAhgMZsRllwBGcJDCAAyLXhBIAuKGSJw8AcwA0TZm3xFSZab2SkILPCroBfANwMGoSLEQoAcslwFi5KrUbMo7Ttz4ChUXESAH5pWUJ5ZVUodRctMKg8AFcSEmNzS2toZ003angkNAwcDVcKOlsix1zyiytwaAAFYjAKanpVAHdgADNwuUUVZgA6MdqtBlNLBjjZKAAPaRbkNvcaKB7+qAAiTAkSZB2oTLngKBBl1vbaKDGRxcyZoA

Use case here is to implement a wrapper component which simply passes through props to Video. Possible solution here could be to simplify the Prop types of the Video component to remove ambiguity for the type checker.

**Comment by rahilvora** [comment]
Can I try this?

**Comment by christianvuerings** [comment]
@pfarejowicz would you be able to have a look at this?

**Comment by pfarejowicz** [comment]
I'm inclined to say that this might be an issue with Flow's ability to make a type inference and not necessarily our code. I agree that simplifying the props to avoid the ambiguity would work to get rid of this error but then it means making all 6 of the a11y labels required by default which I think is not needed especially since they only show up if `controls = true`

**Comment by keyworks** [comment]
I dunno if there are any other flow gymnastics you can do to fix this (I tried everything I could.). 

Here are some other options that have some compromises: 
1. Make accessibility label required in both `controls = null` and `controls = true` cases. It will make things redundant for `controls = null` case but at least we'll have safe flow typings.
2. Make accessibility labels optional for all cases but move validation to runtime (e.g. PropTypes). 

Option 1 is more type safe. Option 2 is more convenient but perhaps less safe since people can ignore the PropType errors.

**Comment by keyworks** [comment]
Addressing this in #321.

**Comment by pfarejowicz** [comment]
Fixed in #321
